packages/components/src/list/collapsible-list/index.tsx

```diff
@@ -0,0 +1,109 @@
+/**
+ * External dependencies
+ */
+import { Dashicon } from '@wordpress/components';
+import { useState, useCallback, Children } from '@wordpress/element';
+import { Transition } from 'react-transition-group';
+
+/**
+ * Internal dependencies
+ */
+import { ExperimentalListItem } from '../experimental-list-item';
+import { ListProps, ExperimentalList } from '../experimental-list';
+
+type CollapsibleListProps = {
+	hideText: string;
+	showText: string;
+	collapsed?: boolean;
+	minChildrenToShow?: number;
+	onCollapse?: () => void;
+	onExpand?: () => void;
+} & ListProps;
+
+const TASK_HEIGHT = 70;
```

Comment:
This arbitrary height works well for single line text that uses whatever style we currently have, but breaks down pretty quickly when more content or markup is included in the `<ExperimentalListItem>`s. Also, there's nothing stopping individual list items all having differing heights.

Is it possible to conditionally render the would-be-hidden list items and perhaps not have to calculate the height value for transitions?

Reply:
Yeah that makes sense, I updated it to calculate the height on the fly, as there is no other way to do the transition nicely without a calculated height. See changes here: https://github.com/woocommerce/woocommerce-admin/pull/6869/commits/319c48ce3837dd334b6a8802ae5b0014df4f9890 it is working quite well.